server/src/main/java/org/opensearch/discovery/InitializeExtensionsRequest.java

```diff
@@ -56,21 +65,41 @@ public DiscoveryNode getSourceNode() {
         return sourceNode;
     }
 
+    public int getPort() {
```

Comment:
Given that it is implied that the port is a "default" that can be changed, should we also have a setter?

Reply:
This is the port of OpenSearch node which is 9200. Not sure having a setter makes sense here.

Reply:
> This is the port of OpenSearch node which is 9200. Not sure having a setter makes sense here.

But the opensearch port [is configurable](https://github.com/opensearch-project/OpenSearch/blob/ec47a937f3730f2af08531cdc9571f3df57fd1cb/distribution/src/config/opensearch.yml#L57-L59) and it is possible to set a different value using the stream input constructor, so why not allow a different value using the non-stream constructor followed by a setter (or adding another overloaded constructor with an `int port` argument)?